Add support executing circuits with a SamplerV2 instance (#1470)
qiskit-community/qiskit-experiments: docs/howtos/runtime_sessions.rst

Use Experiments with Runtime sessions and sampler
=================================================

Problem
-------

> You want to run experiments with a custom `SamplerV2
> <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.SamplerV2>`_ service. 

Reviewer: I would reference the more general class:

```suggestion
You want to run experiments with a custom :class:`qiskit.primitves.SamplerV2` instance. 
```

Author: That would go to `basesamplerv2` but I can change it

.. note::

>     All jobs, by default, run using the ``SamplerV2`` service. When calling ``exp.run`` a 
>     ``SamplerV2`` object will be automatically generated from the specified backend.

Reviewer: ```suggestion
    All jobs, by default, run using the :class:`qiskit_ibm_runtime.SamplerV2` class. When calling ``exp.run`` a 
    ``SamplerV2`` object will be automatically generated to wrap the specified backend.
```

Solution
--------

In this example, we will pass in a ``SamplerV2`` object to a tomography experiment.

.. note::
    If a sampler object is passed to ``exp.run`` then the `run options 
    <https://docs.quantum.ibm.com/api/qiskit-ibm-runtime/qiskit_ibm_runtime.options.SamplerExecutionOptionsV2>`_ of the 
    sampler object are used. The execution options set by the experiment are ignored.

.. jupyter-input::

    from qiskit_ibm_runtime import SamplerV2 as Sampler
    from qiskit_experiments.library.tomography import ProcessTomography
    from qiskit import QuantumCircuit

    service = QiskitRuntimeService(channel="ibm_quantum")
    backend = service.backend("ibm_osaka")
    qc = QuantumCircuit(1)
    qc.x(0)

    sampler = Sampler(backed)
    # set the shots in the sampler object
    sampler.options.default_shots = 300
    exp = ProcessTomography(qc)
    # Artificially lower circuits per job, adjust value for your own application
    exp.set_experiment_options(max_circuits=3)
    # pass the sampler into the experiment

>     exp_data = exp.run(sampler)

Reviewer: `backend` is still the first argument to `run()`:

```suggestion
    exp_data = exp.run(sampler=sampler)
```

    # This will prevent further jobs from being submitted without terminating current jobs
    backend.close_session()

> 

Reviewer: Can we delete this whole session part? `backend.session` is also deprecated and the note about needing at least qiskit-ibm-runtime 0.15 feels pretty dated when 0.31 is out. It might be nice to keep some mention that circuits will be executed using the sampler's `run()` method that could be run inside of a session.
